lib/Login/SocialLogin.class.inc

```diff
@@ -1,4 +1,35 @@
 <?php
+// Autoloader for the new Facebook API
+require_once(LIB . 'External/Facebook/autoload.php');
+
+// It would be cleaner to store the facebook CSRF token in the session but this will do.
+if(!session_id()) {
```

Comment:
I see some very similar code to this in the twitter code - it's probably best to consolidate the code between the two